docs/howto/local-checkouts.md

```diff
@@ -0,0 +1,343 @@
+| **pull**                                 | Fetches latest changes from a lakeFS remote location into a connected local directory                                                               |                                                                                                                                                                                                                           |
+| **checkout**                             | Syncs a local directory with the state of a lakeFS ref                                                                                              |                                                                                                                                                                                                                           |
+
+For additional information see the [CLI reference](../reference/cli.md#lakectl-local). 
+
+**Note:** The data size you work with locally should be reasonable for smooth operation on a local machine.
+{: .note }
+
+## Example: Using _lakectl local_ in tandem with Git 
+
+We are going to develop an ML model that predicts whether an image is an Alpaca or not. Our goal is to improve the input 
+for the model. The code for the model is versioned by Git while the model dataset is versioned by lakeFS. We will
+be using lakectl local to tie code versions to data versions to achieve model reproducibility.  
+
+### Setup
+{: .no_toc}
+
+To get start with, we have initialized a Git repo called `is_alpaca` that includes the model code:
+![Git repo](../assets/img/lakectl-local/code_repo.png)
+
+We also created a lakeFS repository and uploaded the _is_alpaca_ train [dataset](https://www.kaggle.com/datasets/sayedmahmoud/alpaca-dataset) 
+by Kaggel into it:
+![lakeFS repo](../assets/img/lakectl-local/lakefs-repo-with-train-dataset.png)
+
+### Create an Isolated Environment for Experiments
+{: .no_toc}
+
+Our goal is to improve the model predictions. To meet our goal, we will experiment with editing the training dataset. 
+We will run our experiments in isolation to not change anything until after we are certain the data is improved and ready.    
+
+Let's create a new lakeFS branch called `experiment-1`. Our _is_alpaca_ dataset is accessible on that branch, 
+and we will interact with the data from that branch only.
+![experiment-1-branch](../assets/img/lakectl-local/experiment-branch.png)
+
+
+On the code side, we will create a Git branch also called `experiment-1` to not pollute our main branch with a dataset
+which is under tuning.
+
+### Clone lakeFS Data into a Local Git Repository  
+{: .no_toc}
+
+Inspecting the `train.py` script, we can see that it expects an input on the `input` directory.
+
+<img src="../assets/img/lakectl-local/train_script_internals.png" alt="Expected input location" width="600"/>
```

Comment:
Avoid screenshots of code, they can't be searched and are hard to modify later...